### PR TITLE
Clarify how to adjust GP hyp masks and cutoffs

### PR DESCRIPTION
**Pull request by stevetorr**
Codifying some advice @nw13slx has given me in private communication into FLARE documentation, and updating the GP update_kernel method to simplify changing an extant GP to use a new set of hyperparameters.

### CODE REVIEW

**Comment by stevetorr** [comment]
@nw13slx feel free to squash and merge if this looks good!

File: flare/gp.py [hunk 1]
```diff
@@ -213,14 +219,21 @@ def check_instantiation(self):
...
             kernels, component, hyps_mask)
         self.kernel = kernel
         self.kernel_grad = grad
         self.energy_force_kernel = efk
         self.energy_kernel = ek
         self.kernels = kernel_str_to_array(kernel.__name__)
+        
+        if hyps_mask is not None:
+            self.hyps_mask = hyps_mask
+            if self.cutoffs!= hyps_mask['cutoffs']:
```
Comment (nw13slx) [inline flare/gp.py:233]: maybe make cutoff as an argument instead? hyps_mask does not necessarily have a cutoff argument.

File: flare/gp.py [hunk 2]
```diff
@@ -213,14 +219,21 @@ def check_instantiation(self):
...
+                self.adjust_cutoffs(hyps_mask['cutoffs'],train = False,
+                                   new_hyps_mask = hyps_mask)
+            self.hyps = hyps_mask['hyps']
```
Comment (nw13slx) [inline flare/gp.py:236]: same as above. let's just make hyps as an optional input argument.

File: flare/utils/parameter_helper.py [hunk 3]
```diff
@@ -93,6 +93,15 @@
...
 See more examples in functions ``ParameterHelper.define_group`` , ``ParameterHelper.set_parameters``,
 and in the tests ``tests/test_parameters.py``
+
+If you want to add in a new hyperparameter set to an already-existing GP, you can perform the 
+following steps:
+
+>> hyps_mask = pm.as_dict()
+>> hyps = hyps_mask['hyps']
+>> kernels = hyps_mask['kernels']
+>> gp_model.update_kernel(kernels, 'mc', hyps_mask)
```
Comment (nw13slx) [inline flare/utils/parameter_helper.py:103]: can be revised accordingly.

**Pull request merged**